lib/connectors/base/connector.rb

```diff
@@ -12,27 +12,38 @@
 module Connectors
   module Base
     class Connector
+      def initialize
+        @status = {
+          :index_document_count => 0,
+          :deleted_document_count => 0,
+          :error => nil
+        }
+      end
+
       def sync_content_and_yield(connector)
-        error = nil
         sync_content(connector)
       rescue StandardError => e
         Utility::Logger.error_with_backtrace(message: "Error happened when syncing #{display_name}", exception: e)
-        error = e.message
+        @status[:error] = e.message
       ensure
-        yield error
+        yield @status.dup
       end
 
       def sync_content(connector)
-        error = nil
         sync(connector)
       rescue StandardError => e
         Utility::Logger.error("Error happened when syncing #{display_name}. Error: #{e.message}")
-        error = e.message
+        @status[:error] = e.message
       ensure
-        yield error
+        yield @status.dup
       end
 
-      def sync(connector); end
+      def sync(connector)
+        @sink = Utility::Sink::CombinedSink.new(
+          [Utility::Sink::ConsoleSink.new,
+           Utility::Sink::ElasticSink.new(connector['index_name'])]
+        )
```

Comment:
I moved this to `Connectors::Base::Connector`, if any of its subclasses want to add more, it can do so in its own `sync` method

Reply:
I'm thinking we should add a separate overridable method for this, but we can address this later if needed.